cpp/src/parquet/arrow/arrow_reader_writer_test.cc

```diff
@@ -5146,5 +5147,96 @@ TEST(TestArrowReadWrite, FuzzReader) {
   }
 }
 
+TEST(TestArrowReadWrite, WriteReadPageIndexRoundTrip) {
+  // Enable page index to the writer.
+  auto writer_properties = WriterProperties::Builder()
+                               .enable_write_page_index()
```

Comment:
it would be nice to add an assertion to at least one other test to confirm the page index is in fact not generated.

Reply:
Added a check here: https://github.com/apache/arrow/pull/34054/files#diff-4bc4a077197b0bc71088567adb45a3a20e850290d76cb7af52e2f6b8c19ff6ebR5091